Make task abstract

potatowagon opened the pull request:
20->19 instantiating of Task
Parser: remove the need to return new AddCommand, which instantiates Task.
Parser: when add is entered as input, switch case calls prepareCommand(arguments). Removed `return new AddCommand` from `prepareCommand` method, and enable `prepareCommand` method to identify the type of task based on number of arguments entered, and then from there return new the specific type of parser. This change allows the code to be a step closer to userguide 

prev command input format:
`task` 
`due`
`event`

now:
`add`

Commit:
Getting started
Task class: make abstract

Commit:
Parser: change command format to depend on add only. Change prepareAdd method, which initiates AddCommand which initiates Task, to call add(Event/Float/Deadline)Parser. One less line initiating Task

pyokagan commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -18,22 +18,9 @@
+	public CommandResult execute() {
+		// TODO Auto-generated method stub
+		return null;
+	}
```
This change makes AddCommand effectively useless, so why not just delete it altogether?

potatowagon replied:
ried but failed, because that is an abstract method in Command

pyokagan replied:
@potatowagon I meant, delete the whole AddCommand.java

potatowagon replied:
ok

potatowagon replied:
But Parser depends on AddCommand.COMMAND_WORD, which is add. I have deleted the AddCommand file and replaced AddCommand.COMMAND_WORD in Parser with "add", though i am unsure if this will violate code quality

pyokagan replied:
Right, but you need to do more than just blind removal of code. You need to think about why the code you are removing even exists in the first place.

Q: Why does `Parser` depend on `AddCommand.COMMAND_WORD`?

A: Because it parses the `add` command, which then returns an instance of `AddCommand`.

Q: What happens when the new instance of `AddCommand` is returned?

A: The `LogicManager` class calls `execute()` on it.

Q: With the changes in this PR, what happens when `execute()` is called?

A: Nothing happens, and `null` is returned.

Q: But does the `LogicManager` even expect `null` to be returned? What happens if it gets a `null`? ......

A: Well, likely a `NullPointerException` will occur somewhere or `null` may appear on the command result display... anyway the point is that `AddCommand` now does not do anything remotely useful (and in fact, I think, it may cause some tests to fail). As such, perhaps `AddCommand`, and the code in the `Parser` that parses the `add` command, should be completely removed.

pyokagan commented:
The commit [2/3] [Parser: change command format to depend on add only. Change prepareAdd method, which initiates AddCommand which initiates Task, to call add(Event/Float/Deadline)Parser. One less line initiating Task](https://github.com/CS2103AUG2016-T11-C4/main/pull/73/commits/2a6f9b380a0561fe95f0bfcdaa1998fc3a41e29c) tries to do two things at once:
1. Remove AddCommand's dependency on `Task`
2. Change the parsing logic used in `Parser`.

Only `(1)` is relevant to the goal of the PR, `(2)` should be kept out.

Commit:
19->7 instantiating of task

TestUtil: Replace Task instances with FloatingTask instances
LogicManagerTest: Replace Task instances with FloatingTask instances

Commit:
7->0 instances of initiate task

TaskTrackerGuiTask: replace new Task(Name name) with new FloatingTask(String name)
 illegal excetion handlers

pyokagan commented:
@potatowagon Only use the `ReadyForReview` label when you are actually done. Like, fully complete.

Commit:
Reversing changes done in Parser so pull request for Make Task Abstract only makes task abstract

Commit:
Reversing changes doe to parser, delete implementations to use `add` command only instead of three different comands

Commit:
Parser: delete the AddCommand file and replaced AddCommand.COMMAND_WORD in Parser with "add"

Commit:
remove code for add command, and prepreAdd() which is not used now that add command is gone

pyokagan reviewed (changes requested):
Why are spaces being replaced with tabs for indentation? This makes the diff completely useless (see the "Files Changed" of this PR), and I find it extremely hard to see the changes in this PR and review it.

potatowagon commented:
is tab auto added after a new line enter, i dont know how that happened too

Commit:
resolve checkstyle issues

Commit:
resolve checkstyle issues

Commit:
remove unused imports

pyokagan reviewed (changes requested):
Tests do not pass. **Make sure all tests pass before even adding the `ReadyForReview` label**

pyokagan commented:
This PR is not so thorough in cleansing non-abstract `Task` from the code base, and has been going for quite long already. It's also blocking my logic component rework. I think I'll just handle this.

pyokagan commented:
FWIW, I did try to see if I could merge in any useful bits, but failed because I think the procedure is not so thorough.

potatowagon commented:
ok
